src/firewall/core/nftables.py

```diff
@@ -404,7 +404,8 @@ def build_flush_rules(self):
             rules.append({"delete": {"table": {"family": "inet",
                                                "name": TABLE_NAME}}})
             self.created_tables["inet"].remove(TABLE_NAME)
-        return rules
+
+        self.set_rules(rules, log_denied=log_denied)
```

Comment:
the later commit adds a `try:` around the `self.set_rules()` call.

Even if firewalld tracks some state (to know that it configured the table earlier), there is no guarantee that the table wasn't deleted in the meantime (or any other reason, why the nftables command might fail). It seems that an unchecked except at this point is fatal, as it backs out in the middle of a larger operation.

OK, the next commit fixes that, by adding the try-catch. But what about the other backends? An invocation of iptables may fail at any time, it seems fragile to not catch exceptions. Maybe the set-rules methods shouldn't signal a problem via an exception (return code instead), because it's cumbersome to handle them. Or as they do, we really should always handle `ValueError` for every call.

Reply:
> iptables may fail at any time, it seems fragile to not catch exceptions

The fragile part is, that optimally we would not trust /sbin/iptables or libnftables to always succeed and handle/anticipate failures. Letting an exception escape from deep down during `flush()`, probably messes badly with the internal state of firewalld.

Reply:
What we would we do if this fails? IMO failure to flush() is fatal anyways.

Reply:
ok. then it would be handled by the try/except in `src/firewall/core/fw.py:start()`?

I just find it ugly, that we are in the middle of doing something (internal state was already partly updated, and in the middle something went wrong). It's not clear that backing out in the middle doesn't leave broken state behind. But possibly it does not?

Reply:
In most changes, i.e. transactions, they changes are accepted or rejected in whole. In those case the failure doesn't mangle the internal state.

However, in this case the failure is _expected_ in many cases, i.e. clean startup. IMO it makes sense to handle here.

It's true that a `flush` command could fail, but if it does then the firewall backend is broken and we're beyond hope anyways. I don't see value in adding extra error handling. The startup code will attempt to fallback to safe defaults on error; if the defaults also fail then we shut down completely.